tests/requests/orders/create.test.ts

```diff
@@ -0,0 +1,177 @@
+import 'tests/setup';
+import server from 'src/server';
+import Order from 'src/models/Order';
+import OrderItem from 'src/models/OrderItem';
+import Product from 'src/models/Product';
+import { LightMyRequestResponse } from 'fastify';
+
+describe('POST /orders', () => {
+    describe('when the input is valid', () => {
+        it('when pass all the parameters', async () => {

```

Comment:
a função`it` normalmente define o que tá sendo esperado (expectation) e não o cenário de teste

```suggestion
    describe('when the input is valid', () => {
        it('creates an order and returns 201', async () => {
```

vc pode compor seus cenários utilizando `describes` aninhados:

```js
    describe('when the input is valid', () => {
        describe('when another condition happens', () => {
            it('creates an order and returns 201', async () => {
```